### PR TITLE
isc-dhcp: write resolv.conf per network & dhcp settings

### PR DESCRIPTION
**Pull request by pprindeville**
Maintainer: me
Compile tested: x86_64, generic, HEAD (ed8416f)
Run tested: same

Built `.ipk` and installed it on running production system.  Stopped and restarted `dhcpd` service.  Validated `/tmp/resolv.conf` and `/tmp/run/dhcpd.conf` files for correctness.

Description:

Use the top-level `domain` option value from `/etc/config/dhcp` to generate the default `domain` in `resolve.conf`.  Use the `dns` option value from the `/etc/config/network`'s `interface` sections to generate `nameserver` list.

### CODE REVIEW

**Comment by pprindeville** [comment]
@champtar There are only a couple of names available in the reviewers list to select from.  Any idea why that is?  I'd like to ask a few other people to have a look... like @dedeckeh, but I can't add him to the list.

**Comment by hnyman** [comment]
He was now on the list.

**Comment by pprindeville** [comment]
Thanks, Hannu!

**Comment by pprindeville** [comment]
It's a pretty trivial fix...  I'll merge if no one has any issues with it by tomorrow end of day.

File: net/isc-dhcp/files/dhcpd.init [hunk 1]
```diff
@@ -263,6 +265,11 @@ general_config() {
 	echo "max-lease-time $max_lease_time;"
 
 	[ -n "$domain" ] && echo "option domain-name \"$domain\";"
+
+	rm -f /tmp/resolv.conf
+	echo "# This file is generated by the DHCPD service" > /tmp/resolv.conf
+	echo "domain $domain" >> /tmp/resolv.conf
```
Comment (dedeckeh) [inline net/isc-dhcp/files/dhcpd.init]: Don't you have to check the domain name is non empty before echoing it to resolv.conf ?

Comment (pprindeville) [reply net/isc-dhcp/files/dhcpd.init]: Fixed.

File: net/isc-dhcp/files/dhcpd.init [hunk 2]
```diff
@@ -180,6 +180,8 @@ gen_dhcp_subnet() {
 	echo " option domain-name-servers $DNS;"
 	config_list_foreach "$cfg" "dhcp_option" append_dhcp_options
 	echo "}"
+
+	echo "nameserver $DNS" >> /tmp/resolv.conf
```
[the remark this answers is not in the document]
Comment (pprindeville) [reply net/isc-dhcp/files/dhcpd.init]: So just use 127.0.0.1 then?

**Pull request merged**